### PR TITLE
create integrations index if it does not exist and define the integrations document

### PR DESCRIPTION
**Pull request by derek-ho**
### Description
[Describe what this change achieves]

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Swiddis** [comment]
Is this being repurposed per #1521 or should we close it?

File: src/main/kotlin/org/opensearch/integrations/model/IntegrationObjectDataProperties.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.integrations.model
+
+import org.opensearch.common.io.stream.Writeable
+import org.opensearch.core.xcontent.XContentParser
+
+internal object IntegrationObjectDataProperties {
```
Comment (YANG-DB) [inline src/main/kotlin/org/opensearch/integrations/model/IntegrationObjectDataProperties.kt:11]: missing the integration actual fields - see [here](https://github.com/opensearch-project/observability/blob/main/src/main/resources/schema/system/integration.schema)

File: src/main/kotlin/org/opensearch/integrations/model/IntegrationObjectDoc.kt [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+ * Data class representing IntegrationsObject.
+ */
+data class IntegrationObjectDoc(
+    val objectId: String,
+    val updatedTime: Instant,
+    val createdTime: Instant,
+    val tenant: String,
+    val access: List<String>, // "User:user", "Role:sample_role", "BERole:sample_backend_role"
+    val type: IntegrationObjectType,
+    val objectData: BaseObjectData?
```
Comment (YANG-DB) [inline src/main/kotlin/org/opensearch/integrations/model/IntegrationObjectDoc.kt:37]: why not having the concrete ObjectDataProperties here?

**Review by YANG-DB — APPROVED** [review]
I've added few comments - lets review them after we conclude our demo and think again on the design

**Comment by derek-ho** [comment]
Closing since it moved to the front end

**Pull request closed without merging**